File: LeetCode/1869. Longer Contiguous Segments of Ones than Zeros/solution.py
class Solution:
    def checkZeroOnes(self, s: str) -> bool:
        ones, zeros, cur = 0, 0, 0
        n = len(s)
        for i in range(n):
            if i and s[i] == s[i-1]:
                cur += 1
            else:
                cur = 1
            if s[i] == '1':
                ones = max(ones, cur)
            else:
                zeros = max(zeros, cur)
        return ones > zeros

File: LeetCode/1869. Longer Contiguous Segments of Ones than Zeros/solution.cpp
class Solution {
public:
    bool checkZeroOnes(string s) {
        int n = s.size();
        vector<int> a(2);
        for (int i = 0, j; i < n; i = j) {
            for (j = i + 1; j < n && s[j] == s[i]; ++j);
            int x = s[i] - '0';
            a[x] = max(a[x], j - i);
        }
        return a[1] > a[0];
    }
};


class Solution {
public:
    bool checkZeroOnes(string s) {
        int mx1=0, mx0=0;
        int cur1=0, cur0=0;
        for (char c: s) {
            if (c=='1') {
                cur1++;
                cur0=0;
            } else {
                cur1=0;
                cur0++;
            }
            mx1=max(mx1, cur1);
            mx0=max(mx0, cur0);
        }
        return mx1>mx0;
    }
};